communications/mqtt/publish/publish.ino

```diff
@@ -0,0 +1,110 @@
+#include <Wire.h>
+#include <WiFi.h>
+#include <PubSubClient.h>
+#include <ArduinoJson.h>
+
+// Replace the next variables with your Wi-Fi SSID/Password
+const char* WIFI_SSID = "WIFI_SSID";
+const char* WIFI_PASSWORD = "WIFI_PASSWORD";
+char macAddress[18];
+
+// Add MQTT Broker settings
+const char* MQTT_BROKER_IP = "BROKER_IP";
+const int MQTT_PORT = 1883;
+const bool RETAINED = true;
+const int QoS = 0;  // Quality of Service for the subscriptions
+
+WiFiClient espClient;
+PubSubClient client(espClient);
+
+// Topic Variables
+String jsonTopicStr;
+String newTopicStr;
+
+
+void setup() {
+  Serial.begin(9600);  // Starts the serial communication
+  Serial.println("");
+
+  client.setServer(MQTT_BROKER_IP, MQTT_PORT);  // Connect the configured mqtt broker
+  //client.setCallback(callback);  // Prepare what to do when a message is recieved
+
+  connectToWiFiNetwork();  // Connects to the configured network
+  connectToMqttBroker();  // Connects to the configured mqtt broker
+  
+}
+
+void loop() {
+  checkConnections();
+  publishMqttData();  // Publishes to counter topic
+  publishMqttJson();  // Subscribes to json topic
+  
+  delay(7000);
+}
+
+/* Additional functions */
+void publishMqttData(){
+  String Data = "Hello there! \n This is a message from my ESP32 with MacAdress:" + String(macAddress)+"\n";
+  newTopicStr = String(macAddress) + String("/test");
+  const char* newTopic = newTopicStr.c_str();
+  const char* PubData = Data.c_str();
+  client.publish(newTopic, PubData);
+  Serial.println("Client MQTT published to topic: " + String(newTopic) + " (QoS:" + String(QoS) + ")");
+}
+
+void publishMqttJson(){
+  char buffer[512];  //create the buffer where we will print the JSON document to publish through MQTT
+  
+  //Create JSON document
+  StaticJsonDocument<300> JSONdoc; // a little more than 300 bytes in the stack
+  JSONdoc["device"] = "ESP32"; //add names and values to the JSON document
+  JSONdoc["sensorType"] = "Temperature";
+  JsonArray values = JSONdoc["values"].to<JsonArray>(); //Or we can add an array to the string "values"
+  
+  values.add(27);  //Inside the array we can add new values to "values" 
+  values.add(29); 
+  
+  serializeJson(JSONdoc, buffer); //serialize the JSON document to a buffer in order to publish it
+
+  jsonTopicStr = String(macAddress) + String("/json");
+  const char* jsonTopic = jsonTopicStr.c_str();
+  client.publish(jsonTopic, buffer);
```

Comment:
if we are using `QoS` then we need to add it also in the `client.publish()` function

Reply:
I see that the publish does not support QoS https://pubsubclient.knolleary.net/api.html#publish1